platform-jobs/samza/collection-migration/src/main/java/org/ekstep/jobs/samza/service/CollectionMigrationService.java

```diff
@@ -28,324 +31,428 @@
-			children = children.stream().map(child ->  {
-				if (StringUtils.equalsIgnoreCase((String) child.get("mimeType"), "application/vnd.ekstep.content-collection") && StringUtils.equalsIgnoreCase((String) child.get("visibility"), "Parent")) {
-					idsToDelete.add((String) child.get("identifier"));
-				}
-				return child;
-			}).collect(Collectors.toList());
-			List<Map<String, Object>> nextChildren = children.stream()
-					.map(child -> (List<Map<String, Object>>) child.get("children"))
-					.filter(f -> CollectionUtils.isNotEmpty(f)).flatMap(f -> f.stream())
-					.collect(Collectors.toList());
-			if (CollectionUtils.isNotEmpty(nextChildren)) {
-				getCollectionIdsToDelete(nextChildren, idsToDelete);
-			}
-		} else {
-			LOGGER.info("Children is empty: "+ children);
-		}
-	}
+                List<Node> nodesForUpdate = new ArrayList<>();
+                nodesForUpdate.add(ecmlNode);
+                if (isImage)
+                    nodesForUpdate.add(ecmlImageNode);
+                //If no media contains drive urls then update node version
+                if (CollectionUtils.isEmpty(contentUrls)) {
+                    migrationService.updateEcmlNode(nodesForUpdate);
+                    return;
+                }
+                //Get all the contents whose drive urls don't have an existing asset
+                Set<String> contentUrlsWithNoAsset = new HashSet<>();
+                Map<String, String> driveArtifactMap = new HashMap();
+                contentUrls.forEach(contentUrl -> {
+                    String artifactUrl = migrationService.doesAssetExists(contentUrl);
+                    if (StringUtils.isBlank(artifactUrl))
+                        contentUrlsWithNoAsset.add(contentUrl);
+                    else
+                        driveArtifactMap.put(contentUrl, artifactUrl);
+                });
+                //Download the content, create asset and upload the respective content
+                Map<String, List> fileMap = new HashMap();
+                contentUrlsWithNoAsset.forEach(contentUrl -> fileMap.putAll(migrationService.downloadDriveContents(contentUrl, downloadFileLocation)));
+                fileMap.keySet().forEach(driveUrl -> {
+                    String id = migrationService.createAsset(fileMap, driveUrl);
+                    String artifactUrl = migrationService.uploadAsset((File) fileMap.get(driveUrl).get(0), id);
+                    if (StringUtils.isNotBlank(artifactUrl))
+                        driveArtifactMap.put(driveUrl, artifactUrl);
+                });
+                //Delete all the downloaded content
+                migrationService.deleteDownloadedContent(downloadFileLocation);
+                //Update the ecml body and the nodes
+                if (MapUtils.isNotEmpty(driveArtifactMap)) {
+                    if (isImage)
+                        migrationService.ecmlBodyUpdate(contentBody, contentId + ".img", driveArtifactMap);
```

Comment:
contentBody should be replaced with imageContentBody

Reply:
Done Lead Sir